### ISSUE TITLE
Subject line parsing issues - Duplicati adding \r\n?

### ISSUE THREAD

**Issue by ninja6o4** [HandyGuySoftware/dupReport#80]
I've setup dcurrey's method of subject line parsing, and I know that spaces are not supported in the source/dest devices, but this is cause for a problem even when I revert it to no spacing.

```
EmailServer.process_message()
Subject=[Duplicati Backup (Success) report for

 xxxSurgeryCentre_BackRoom-BackupServer]
srcregex=[(?<=report.for.).*\-]  destRegex=[\-.*]
srcdestdelimiter [-] not found in subject. Skipping message.
dremail.getNextMessage()
```

You can see it's seeing carriage returns to the subject line.
Here is the raw subject line when it pulls the message:
`Subject: Duplicati Backup (Success) report for\r\n xxxSurgeryCentre_BackRoom-BackupServer`

My .rc file is as follows:
```
subjectregex = ^Duplicati Backup \((Success|Error|Warn)\) report for
srcregex = (?<=report.for.).*
destregex = .*
srcdestdelimiter = -
```

**Comment by ninja6o4** [comment]
Actually, viewing the subject line got me thinking, and I'm trying Marc_Aronson's suggested solution of entering the mail settings directly as text in the Settings menu, which hopefully eliminates the erroneous \r\n code that was inserted into the Subject line.

edit: nope, didn't fix it .. Why is Duplicati inserting this \r\n code? This is the subject line being sent:
`--send-mail-subject=Duplicati %OPERATIONNAME% (%PARSEDRESULT%) report for %backup-name%`

**Comment by dcurrey** [comment]
This is an smtp length issue creating your problem.  The subject header line can only be 78 characters then it folds for any additional length.

It seems if you shorten your backup name it will work correctly.

I was able to get it to work with just 123SurgeryCentreBR-BackupServer

This maybe something that can be filtered out.

**Comment by ninja6o4** [comment]
Ahh that explains why when I was using spaces, it would randomly cut off parts of the destination server name, as it would snip it randomly depending on the source computer's name length!

**Comment by dcurrey** [comment]
Try adding this around line 337 in the dremail.py
~~~
            msgParts['subject'] = msgParts['subject'].strip( '\n' )
~~~

The Block should look like this.
~~~
        # get Subject
        decode = email.header.decode_header(msg['Subject'])[0]
        msgParts['subject'] = decode[0]
        if (type(msgParts['subject']) is not str):  # Email encoded as a byte object - See Issue #14
            msgParts['subject'] = msgParts['subject'].decode('utf-8')
            msgParts['subject'] = msgParts['subject'].strip( '\n' )
        globs.log.write(3, 'Subject=[{}]'.format(msgParts['subject']))
~~~
I did a quick check and it seemed to work.  Showed up in the report correctly anyway.
![image](https://user-images.githubusercontent.com/33069071/35421512-9f759c74-0211-11e8-8d43-f401c6d58b6a.png)

**Comment by HandyGuySoftware** [comment]
@ninja6o4 , let me know if this works for you. I'm just finishing up the beta code for 2.2 (hope to release it in the next day or so) and I can slip this in.

**Comment by dcurrey** [comment]
Ummm I may have forgot to change my regex to include the (success)  in my test. Increasing the subject lenght past the 78.

Will have to test this tomorrow.  Sorry about that if it doesn't work.

**Comment by ninja6o4** [comment]
Thanks for the suggestion! However, for some reason I'm getting an error when trying to run the script:
```
Traceback (most recent call last):
  File "dupReport.py", line 23, in <module>
    import dremail
  File "C:\dupReport-2.1.0\dremail.py", line 337
    msgParts['subject'] = msgParts['subject'].strip('\n')
                                                        ^
TabError: inconsistent use of tabs and spaces in indentation
```

**Comment by dcurrey** [comment]
Ok just checked it appears to work correctly with the subjectregex set correctly.


![image](https://user-images.githubusercontent.com/33069071/35438948-b82295f6-0266-11e8-9f75-3a456823481c.png)

Verified the subject header is using multiple lines
~~~
Date: Thu, 25 Jan 2018 16:37:06 -0500
Subject: Duplicati Backup (Success) report for
 123SurgeryCentreBackroom-BackupServer
Message-Id: <K3745BH9H3U4.NF4Z84FKFDWP@DDC3>
~~~

**Comment by HandyGuySoftware** [comment]
Code to handle this was added to the 2.2 beta, released to pre_prod last night.

**Comment by ninja6o4** [comment]
OK so I tried adding that line of code in and it didn't work for some reason, still seems to think it's not a valid message. I tried with ( '\n' ) and with ( '\r\n' ) but no joy.

I updated to pre_prod branch, so the logs look different now:
```
Server.fetch(): retVal=[OK] data=[[(b'74 (BODY[HEADER.FIELDS (DATE SUBJECT MESSAGE-ID)] {194}', b'Date: Thu, 25 Jan 2018 22:06:28 -0800\r\nSubject: Duplicati Backup (Success) report for 123 Surgery Centre _\r\n Reception-Backup Server\r\nMessage-Id: <RSWPG30CH3U4.4JQ4WKP16IPZ@Sheila-PC>\r\n\r\n'), b')']]
dataSpl=[['Date: Thu, 25 Jan 2018 22:06:28 -0800', 'Subject: Duplicati Backup (Success) report for 123 Surgery Centre _', ' Reception-Backup Server', 'Message-Id: <RSWPG30CH3U4.4JQ4WKP16IPZ@User-PC>', '']]
srcregex=[(?<=report.for.).*\-]  destRegex=[\-.*]
SrcDestDelimeter [-] not found in subject. Skipping message.
```

```
Server.fetch(): retVal=[OK] data=[[(b'75 (BODY[HEADER.FIELDS (DATE SUBJECT MESSAGE-ID)] {188}', b'Date: Thu, 25 Jan 2018 22:35:28 -0800\r\nSubject: Duplicati Backup (Success) report for\r\n 123SurgeryCentre_BackRoom-BackupServer\r\nMessage-Id: <CV03438CH3U4.JZU91C3UXEMQ@User2-PC>\r\n\r\n'), b')']]
dataSpl=[['Date: Thu, 25 Jan 2018 22:35:28 -0800', 'Subject: Duplicati Backup (Success) report for', ' 123SurgeryCentre_BackRoom-BackupServer', 'Message-Id: <CV03438CH3U4.JZU91C3UXEMQ@User2-PC>', '']]
srcregex=[(?<=report.for.).*\-]  destRegex=[\-.*]
SrcDestDelimeter [-] not found in subject. Skipping message.

```

**Comment by HandyGuySoftware** [comment]
OK, I see what's going on here (and it's not good).

TL;DR: The new email parsing routines **_really_** do not like having extra \n's in the subject line!

The long version: The new parser extracts just the headers first, to see if it's a message of interest. This saves download time for emails that are not MoI. The Python IMAP libraries pull the headers as a \n-separated string, then dR splits it up assuming that the \n is the end of the header segment. Since your subject has extra \n's there, it's taking that as a new header segment, and bad things happen from there (as you can see).

I haven't checked for sure, but I'm pretty sure the 2.1 code would have the same problem. So, anything longer than 78 characters would cause an error.

I'll need to really take a look at this tonight to find a work-around, because if a \n is not always the border between header segments, that blows dR's whole approach to header parsing. In the meantime, anything you can do to shorten your subject lines will allow you to continue processing with dR. 

Stay tuned...

**Comment by dcurrey** [comment]
Yea the code I added doesn't really do anything with dupReport.   It will indeed process the mail correctly with the return in the subject header.   

What version of duplicati are you running and any info on the email server might help.

Can you you supply an actual copy of your email headers?   My server doesn't show them on the long lines just listed like above.

Not sure if this would help I was able to get around the problem with the default srcregex=\w* and destregex=\w*  I was able to process them correctly with the defaults.  But then my email server isn't adding the \r\n literally into the subject header.   I am somewhat unclear if this is happening or if dupreport itself is just reporting this in the logs.

edit:  This is still being tested on the 2.1 code

**Comment by ninja6o4** [comment]
I can get the headers when I'm back at my desk. I'm running the latest
duplicati v2 beta, and the mail server is a standard cpanel based mail
server.

**Comment by dcurrey** [comment]
Just checked out the new code.   Even if the changes I mentioned above with the src and dest regex change it still will not work on dupreport 2.2.   As handyguy mentions above the way emails are processed is different.   The return in middle of subject flat out kills it.

Its above my ability to find some kind of fix.

**Comment by ninja6o4** [comment]
Here are the headers:
```
Return-Path: <omega@mydomain.com>
Delivered-To: backuplogs@mydomain.com
Received: from bhs1.mydomain.com
	by bhs1.mydomain.com with LMTP id oGf7LjLMalrAYgAA7LBDPQ
	for <backuplogs@mydomain.com>; Thu, 25 Jan 2018 22:35:30 -0800
Return-path: <omega@mydomain.com>
Envelope-to: backuplogs@mydomain.com
Delivery-date: Thu, 25 Jan 2018 22:35:30 -0800
Received: from [000.81.94.106] (port=61355 helo=[192.168.1.113])
	by bhs1.mydomain.com with esmtpsa (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89_1)
	(envelope-from <omega@mydomain.com>)
	id 1eexbm-0006ZV-Ap; Thu, 25 Jan 2018 22:35:30 -0800
From: Duplicati Backup Report <omega@mydomain.com>
Date: Thu, 25 Jan 2018 22:35:28 -0800
Subject: Duplicati Backup (Success) report for
 123SurgeryCentre_BackRoom-BackupServer
Message-Id: <CV03438CH3U4.JZU91C3UXEMQ@User-PC>
To: backuplogs@mydomain.com
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 7bit
```

**Comment by HandyGuySoftware** [comment]
@ninja6o4 & @dcurrey, I think I've worked up a fix for this. If it works, it also solves a couple of other header parsing problems I've had but chosen to ignore 'till now.

Please download the code from the [Issue_80 branch](https://github.com/HandyGuySoftware/dupReport/tree/Issue_80) and see if that fixes the problem. 

Keeping my fingers crossed! :-)

**Comment by dcurrey** [comment]
Looks good on my test server with the long names.

**Comment by ninja6o4** [comment]
Fingers crossed, I can't tell yet as I'm having trouble due to the problem I'm having for issue 81..

**Comment by ninja6o4** [comment]
Confirmed it works here now :) Thanks!

**Comment by HandyGuySoftware** [comment]
I have some clean-up & commenting to do, but I will post this to the pre_prod branch by the end of the day.
